Move default adapter under app/adapters and let resolver resolve the adapter

abuiles opened the pull request:
Removes the need of having window.App defined for specifing an
adapter, now it will be resolved by the resolver. Will also facilitate
things for adding tests with QUnit.

bcardarella commented on lib/generators/ember/bootstrap_generator.rb:
```diff
@@ -29,7 +29,7 @@ def create_application_file
 
       def create_ember_adapter_file
-        template "adapter.js.erb", "#{config_path}/adapter.js.erb"
+        template "adapter.js.erb", "#{app_path}/adapters/application.js.es6.erb"
```
I think this would get changed to `adapters/active_model.js.es6.erb`

bcardarella replied:
this should go into `config_path` not `app_path`

abuiles replied:
@bcardarella we could do that but then we would need to make explicit in the store the adapter name. I would leave it like this, since our default application adapter uses AMS.

abuiles replied:
@bcardarella the reason this is in app_path, is that the resolver looks by default for `app/adapters/application`. We could have it in config/ but then we should declare the namespace manually.

bcardarella replied:
if this is a change we can make in the resolver I would prefer to make it there as the adapters are primarily a configuration concern not business logic

abuiles replied:
@bcardarella we need to allow the adapter to be require from  `app/adapters/application` or `app/adapters/active_model`, if the adapters are into config then it ends up being `config/adapters`, any idea how can we force the namespace for adapters to be under `app` and not `config` (even if the file is under config)? cc @stefanpenner

bcardarella replied:
Nah I disagree with that. Even if that is what they are doing in ember-app-kit we aren't going to do that in eak-rails. Adapters are not business logic. We'll have to modify the resolver to force it to lookup on config. I'll start looking into this.

bcardarella replied:
@abuiles I have this pending PR which will fix the adapter resolving issue. Now adapters will resolve to `config@adapter:x`

@rjackson thoughts?

bcardarella replied:
pending PR: https://github.com/dockyard/ember-appkit-rails/pull/138

rwjblue replied:
In general I am not opposed, but I would much prefer to stay in sync with EAK on basic structural stuff. Perhaps we can push these changes upstream (from #138) so that we can still be in lock-step...

bcardarella replied:
+1 if we can get EAK on board with this structure

abuiles replied:
@bcardarella sounds great then! I'm about to board a plane for 6 hours :S, will update the PR once I land. And agree with @rjackson too, would be great if we are in sync with EAK.

bcardarella replied:
@abuiles update your branch, I just merged in #138

rwjblue commented:
:+1: - also, I believe that this resolves #135.

bcardarella commented on lib/generators/templates/adapter.js.erb:
```diff
@@ -1,3 +1,3 @@
-window.<%= application_name.camelize %>.ApplicationAdapter = DS.ActiveModelAdapter.extend({
+export default DS.ActiveModelAdapter.extend({
```
@stefanpenner does the resolver use es6 for adapters or do ember-data adapters still expect to be set to a constant on the application constant?

abuiles replied:
@bcardarella it uses es6 for adapters. http://cl.ly/image/3R3S0X2e2y0b

bcardarella commented on lib/generators/templates/application.js.erb:
```diff
@@ -2,7 +2,6 @@
 //= require environment
 //= require ember-appkit
 //= require_self
-//= require adapter
```
this would have to change to:

``` js
//= require_tree ./adapters
```

bcardarella commented on test/generators/bootstrap_generator_test.rb:
```diff
@@ -101,7 +101,7 @@ def assert_files(app_path = app_path, config_path = config_path)
     assert_file "#{config_path}/application.js"
     assert_file "#{config_path}/router.js.es6"
-    assert_file "#{config_path}/adapter.js.erb"
+    assert_file "#{app_path}/adapters/application.js.es6.erb"
```
`adapters/active_model_adapter.js.es6.erb`

bcardarella commented on test/support/generator_test_support.rb:
```diff
@@ -52,7 +52,7 @@ def with_config(options = {})
     path = options[:in_path] || app_path
 
-    %W{components templates templates/components routes}.each do |dir|
+    %W{components templates templates/components routes adapters}.each do |dir|
```
this isn't necessary

abuiles commented:
@bcardarella @rjackson Also we should move 

```
  window.App = require('app').default.create({
  });
```

under 

```
 <script>
     window.App = require('app').default.create();
</script>
```

As it is currently it became conflictive when integrating with teaspoon.

abuiles committed:
Move adapter under config//adapters and let the resolver resolve
adapter.

Removes the need of having window.App defined for specifing an
adapter, now it will be resolved by the resolver. Will also facilitate
things for adding tests with QUnit.

abuiles commented:
@bcardarella updated.

Noticed I'm still using `config/adapters/application.js.es6.erb` which is what ember data asks by default `('adapter:application')` and it makes sense since that is what we want as our application adapter. 

If we want to name it `/config/adapters/active_model.js.es6.erb` then we would need to extend the store and specify that the adapter is not going to be the default `application` but `active_model`. A file probably under `config/stores/main.js.es6`

```
 import default DS.Store.extend({
     adapter: 'active_model'
 });
```

I personally would leave it as `application`, that's  what EAK does, but if you want it to be `active_model` I'm fine with that too, we would need to override again the resolver so it looks for `config/stores` not `app/stores`.

bcardarella commented:
@abuiles `application` is good

abuiles commented:
@bcardarella good to go then, let me know if there is something else.
